plugins/pulp_deb/plugins/importers/sync.py

```
@@ -317,20 +317,33 @@ def __init__(self, *args, **kwargs):
         super(SaveDownloadedUnits, self).__init__(*args, **kwargs)
         self.description = _('Save and associate downloaded units')
 
+    def verify_checksum(self, calculated, from_metadata, path):
+        if calculated != from_metadata:
+            raise PulpCodedTaskFailedException(
+                DEBSYNC002, repo_id=self.get_repo().repo_obj.repo_id,
+                feed_url=self.parent.feed_url,
+                filename=os.path.basename(path),
+                checksum_expected=from_metadata,
+                checksum_actual=calculated)
+
     def process_main(self, item=None):
         path_to_unit = self.parent.step_download_units.path_to_unit
         repo = self.get_repo().repo_obj
         for path, unit in sorted(path_to_unit.items()):
-            # Verify checksum first
-            with open(path, "rb") as fobj:
-                csum = unit._compute_checksum(fobj)
-            if csum != unit.checksum:
-                raise PulpCodedTaskFailedException(
-                    DEBSYNC002, repo_id=self.get_repo().repo_obj.repo_id,
-                    feed_url=self.parent.feed_url,
-                    filename=os.path.basename(path),
-                    checksum_expected=unit.checksum,
-                    checksum_actual=csum)
+            checksums = unit.calculate_deb_checksums(path)
+            if unit.sha1:
+                self.verify_checksum(checksums['sha1'], unit.sha1, path)
+            else:
+                unit.sha1 = checksums['sha1']
+            if unit.md5sum:
+                self.verify_checksum(checksums['md5sum'], unit.md5sum, path)
+            else:
+                unit.md5sum = checksums['md5sum']
+            if unit.sha256:
+                self.verify_checksum(checksums['sha256'], unit.sha256, path)
+            else:
+                unit.sha256 = checksums['sha256']
+            self.verify_checksum(unit.checksum, unit.sha256, path)
```

Is this what we want? Do we want to check every checksumtype? Do we want to throw an error as soon as we encounter a broken checksum, or collect any errors from all three types and then throw a single error?

---

I am leaning heavily towards we do want to check every checksumtype. We just don't want broken checksums/files to make it into our data base...
I am unsure how `PulpCodedTaskFailedException` works, so I would have to look that up to make any changes to what is being thrown as an error.